src/utils/passport.ts

```diff
@@ -0,0 +1,44 @@
+import * as passport from 'passport';
+import { OAuth2Strategy as GoogleStrategy } from 'passport-google-oauth';
+import config from 'config';
+import { getRepository } from 'typeorm';
+import { User } from '@entities/User';
+import { PassportStatic } from 'passport';
+
+passport.use(new GoogleStrategy(
+  {
+    clientID: config.GOOGLE_CONSUMER_KEY,
+    clientSecret: config.GOOGLE_CONSUMER_SECRET,
+    callbackURL: `${config.SERVER_URL}/auth/google/callback`,
+  },
+  async (token, tokenSecret, profile, done) => {
+    console.log(profile);
+    const userRepository = getRepository(User);
+    let user = await userRepository.findOne({ googleId: profile.id });
+    if (user === undefined) {
+      if (profile.name === undefined) {
+        return done(new Error('No name information'), null);
+      }
+      console.log(profile);
+      user = await userRepository.save({
+        firstName: profile.name.givenName,
+        lastName: profile.name.familyName,
+        googleId: profile.id,
+        age: 0,
```

Comment:
Is it relevant to keep an age if you can't get it with google auth ?

Reply:
You can get it, but it's a bit more complicated. It's not implemented yet

Reply:
We could probably make it a nullable field for now